util/TorqueLog.java

```diff
@@ -0,0 +1,153 @@
+
+import java.util.HashMap;
+
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj.shuffleboard.WidgetType;
+
+/**
+ * A wrapper around a shuffleboard tab that allows for widgets to be added 
+ * and updated with a single call in the update function. It also handles auto
+ * positioning of widgets on the board with a given size.
+ * 
+ * @author Justus Languell
+ */
+public final class TorqueLog {
+
+    public static final int ROWS = 6, COLUMNS = 10;
+    private final boolean[][] board = new boolean[ROWS][COLUMNS];
+
+    public final String title;
+    private final ShuffleboardTab tab;
+    private final HashMap<String, NetworkTableEntry> keys; 
+
+    /**
+     * Create a new logging table.
+     * 
+     * @param title The title of the table on the dashboard.
+     */
+    public TorqueLog(final String title) {
+        this.title = title;
+        tab = Shuffleboard.getTab(title);
+        keys = new HashMap<String, NetworkTableEntry>();
+    }
+
+    /**
+     * Log a value to the dashboard.
+     * 
+     * @param key The title of the widget.
+     * @param value The value to log in the widget.
+     * @param width The width of the widget in cells.
+     * @param height The height of the widget in cells.
+     * @param type The type of the cell.
+     */
+    public void log(final String key, final Object value, final int width, final int height, final WidgetType type) {
+        if (!keys.containsKey(key)) {
+            final Position pos = calculatePosition(width, height);
+            if (pos == Position.INVALID) {
+                System.out.println("Could not fit element " + key + " in log " + title);
+                return;
+            }
+            final NetworkTableEntry entry = tab.add(key, value)
+                    .withWidget(type)
+                    .withSize(width, height)
+                    .withPosition(pos.x, pos.y)
+                    .getEntry();
+            keys.put(key, entry);
+        }
+        keys.get(key).setValue(value);
+    }
+    
+    /**
+     * Find the next coordinate location that a widget of size (width, height) can fit onto the dashboard.
+     * 
+     * This algorithm is definitely not the most efficient because it kind of checks every position
+     * on the board. However, it is simple and works well enough for our purposes. Since it only runs once
+     * on the constructor with a very small size it doesn't matter at all.
+     * 
+     * @param width The width of the widget in cells.
+     * @param height The height of the widget in cells.
+     * 
+     * @return The next valid position or INVALID if there isn't a valid position.
+     */
+    private Position calculatePosition(final int width, final int height) {
+        for (int i = 0; i < board.length; i++)
+            for (int j = 0; j < board[i].length; j++) {
```

Comment:
You can reduce the search by not checking the extremes we can rule out.
For the i loop, you only need to go to board.length-height
For the j loop, you only need to go to board[i].length-width

Reply:
@Juicestus